pkg/capture/GPPacket_test.go

```diff
@@ -12,7 +12,132 @@
 
 package capture
 
-import "testing"
+import (
```

Comment:
We now finally have test & benchmarks for this crucial part of our detection logic. Interestingly the benchmark revealed some quite interesting caveats:
* About 25% of the benchmark time is spent on the `reset()` method (where we clear all relevant elements of the packet for reuse)
* The methods accessing stuff via the `capture.Packet` interface from `slimcap` seem to actually cause significant overhead (no inlining, even for simple getters, plus interface safety I assume), probably another 25% of the time is spent on these

I won't touch either of these two things as part of this PR, but I'll think about if there's some low hanging fruits (we could e.g. make the capture.Packet a struct that all capture sources have to adhere to via their interface, that way we can still enforce an certain structure while getting rid of the interface overhead)...

Reply:
Raised https://github.com/fako1024/slimcap/issues/1 for the `capture.Packet` part.

Reply:
Mmm. That sounds like a lot when looking at the percentages. Is the direction detection itself a big factor in packet processing or are other factors far more dominating?

My point being: if direction detection makes up for example 10% of the total time spent per packet, then optimizing that part doesn’t yield as much as others.

Happy to discuss this in detail.

Reply:
Well, IMHO another factor is often overlooked when tackling performance issues (or even "regular" bugs): How much effort is it. If I can get, i.d.k., half of the 25% percent by changing a few lines of code in a package that is not yet even marked as stable then that's worth taking a shot. Of course when there's other low hanging fruits they should be tackled first / as well (which was done as part of this PR, see the fixes and the proposal about source port merging for HTTP / HTTPS). That being said, the an overall assessment is definitively due again once we have a release candidate so we can identify if we have any hot paths that can easily be tweaked...